File: app/src/main/java/com/doushi/test/myproject/znet/InterfaceConfig.java
package com.doushi.test.myproject.znet;

import com.lzy.okgo.cache.CacheMode;

import java.util.HashMap;
import java.util.Map;

/**
 * 请求接口配置信息
 *
 * @author xiemy
 * @date 2018/2/28
 */
public class InterfaceConfig {

    /**
     * 请求接口
     */
    public static final String INTERFACE_URL_STR = "INTERFACE_URL_STR";
    /**
     * 配置请求类型 POST/GET
     */
    public static final String INTERFACE_QUERY_TYPE = "INTERFACE_QUERY_TYPE";
    /**
     * 缓存模式,默认Application配置,一些特殊接口可修改此项配置
     */
    public static final String REQUEST_CONFIG = "RequestConfig";

    public static final String INTERFACE_QUERY_TYPE_POST = "POST";
    public static final String INTERFACE_QUERY_TYPE_GET = "GET";
    /**
     * 上传文件类型
     */
    public static final String InterfaceQueryType_POST_UPDATE = "POST_UPDATE";

    /**
     * 用户信息参数类型
     */
    private static final String PARAMS_TYPE = "PARAMS_TYPE";

    public enum HttpHelperTag {

        /**
         * 获取最新一天的干货
         */
        HTTPHelperTag_ToDay,


        //崩溃上报 start
        /**
         * 崩溃上报
         * ReportCrashLog
         */
        HTTPHelperTag_ReportCrashLog,
        //崩溃上报 end

    }

    //start----user---start
    /**
     * 获取最新一天的干货
     */
    private static final String TODAY = "today";
    /**
     * 崩溃上报
     */
    private static final String RX_URL_REPORT_CRASH_LOG = "UiP1/ReportCrashLog";
    //崩溃上报 end===============

    public static Map<String, Object> interfaceConfigByHttpTag(HttpHelperTag httpHelperTag) {

        Map<String, Object> resultMap = new HashMap<>();
        resultMap.put(REQUEST_CONFIG, false);
        StringBuffer urlStr = new StringBuffer();
        urlStr.append(HttpConfig.getRootUrl());

        // 默认GET
        String queryTypeStr;
        switch (httpHelperTag) {
            // 获取最新一天的干货
            case HTTPHelperTag_ToDay:
                //2.1.3	token登录
                urlStr.append(TODAY);
                resultMap.put(INTERFACE_URL_STR, urlStr.toString());
                resultMap.put(REQUEST_CONFIG,
                        new RequestConfig()
                                //不重连
                                .setRetryCount(0)
                                // 不使用缓存
                                .setCacheMode(CacheMode.NO_CACHE));
                break;

            // region ----- 上传崩溃   ------
            //崩溃上报 GET
            case HTTPHelperTag_ReportCrashLog:
                urlStr.append(RX_URL_REPORT_CRASH_LOG);
                resultMap.put(INTERFACE_URL_STR, urlStr.toString());
                resultMap.put(REQUEST_CONFIG,
                        new RequestConfig()
                                //不重连
                                .setRetryCount(0)
                                // 不使用缓存
                                .setCacheMode(CacheMode.NO_CACHE));
                break;
            // endregion ----- 上传崩溃 ----end

            default:
                urlStr = new StringBuffer();
                resultMap.put(INTERFACE_URL_STR, urlStr.toString());
                queryTypeStr = "";
                resultMap.put(INTERFACE_QUERY_TYPE, queryTypeStr);
                break;
        }
        return resultMap;
    }
}

File: app/src/main/java/com/doushi/test/myproject/znet/InterfaceConfig.kt
package com.doushi.test.myproject.znet

/**
 * 接口配置
 *
 * @author xiemy2
 * @date 2019/3/14
 */
object InterfaceConfig {

    internal const val METHOD_POST = "POST"
    internal const val METHOD_GET = "GET"

    enum class HttpHelperTag
    /**
     * API 接口配置
     *
     * @param method 请求类型 post|get
     * @param apiTag 接口api
     * @param rType  地址类型
     */
    constructor(
            /**地址类型(支付/默认)  */
            @param:HttpConfig.RootType @field:HttpConfig.RootType
            private val rType: Int = HttpConfig.API_OPEN,
            /**请求类型 post|get */
            val method: String? = METHOD_GET,
            /**接口api*/
            private val apiTag: String,
            /**网络请求配置api*/
            var config: RequestConfig? = null) {

        /* ------------------------ > 用户  User Start <---------------------- */
        /**
         * 获取最新一天的干货
         */
        TODAY(HttpConfig.SNS_SDK, "today"),
        /**
         * 分类列表
         */
        ARTICLE_CATEGORY_GET_EXTRA_V1(HttpConfig.SNS_SDK, "article/category/get_extra/v1/"),
        /**
         * 分类列表
         */
        NEWS_FEED_V58(HttpConfig.SNS_SDK, "api/news/feed/v58/"),
        /**
         * 分类列表
         */
        NATIVE_FEED_BROW(HttpConfig.SNS_SDK, "wenda/v1/native/feedbrow"),
        /**
         * 崩溃上报
         */
        RX_URL_REPORT_CRASH_LOG(HttpConfig.SNS_SDK, "UiP1/ReportCrashLog"),

        /***获取支付渠道 */
        GET_PAY_CHANNEL(HttpConfig.SNS_SDK, "/paychannel/android"),
        /* ------------------------ > 用户  User Start <---------------------- */


        /* ------------------------ > 用户  User Start <---------------------- */
        /***获取支付渠道 https://api.apiopen.top/todayVideo */
        TODAY_VIDEO(HttpConfig.SNS_SDK, "/todayVideo");
        /* ------------------------ > 用户  User Start <---------------------- */

        /**是否有需要回调到View(前台显示)*/
        var isAttachedView = true

        /**
         * API 接口配置
         *
         * @param apiTag 接口api
         */
        constructor(apiTag: String) : this(METHOD_GET, apiTag)

        /**
         * API 接口配置
         *
         * @param method 请求类型 post|get
         * @param apiTag 接口api
         */
        constructor(method: String, apiTag: String) : this(HttpConfig.API_OPEN, method, apiTag)

        /**
         * API 接口配置
         *
         * @param apiTag 接口api
         * @param rType  地址类型
         */
        constructor(@HttpConfig.RootType rType: Int, apiTag: String) : this(rType, METHOD_GET, apiTag)

        val apiUrl: String
            get() = HttpConfig.getRootUrl(rType) + apiTag
    }
}
